src/utils/index.ts

```diff
@@ -53,6 +52,158 @@ const promptLayerApiRequest = async (body: TrackRequest) => {
   }
 };
 
+const promptLayerTrackMetadata = async (
+  body: track.Metadata
+): Promise<boolean> => {
+  try {
+    const response = await fetch(
+      `${URL_API_PROMPTLAYER}/library-track-metadata`,
+      {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          ...body,
+          api_key: getApiKey(),
+        }),
+      }
+    );
+    const data = await response.json();
+    if (response.status !== 200) {
+      warnOnBadResponse(
+        data,
+        "WARNING: While tracking your metadata PromptLayer had the following error"
+      );
+      return false;
+    }
+  } catch (e) {
+    console.warn(
+      `WARNING: While tracking your metadata PromptLayer had the following error: ${e}`
+    );
+    return false;
+  }
+  return true;
+};
+
+const promptLayerTrackScore = async (body: track.Score): Promise<boolean> => {
+  try {
+    const response = await fetch(`${URL_API_PROMPTLAYER}/library-track-score`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        ...body,
+        api_key: getApiKey(),
+      }),
+    });
+    const data = await response.json();
+    if (response.status !== 200) {
+      warnOnBadResponse(
+        data,
+        "WARNING: While tracking your score PromptLayer had the following error"
+      );
+      return false;
+    }
+  } catch (e) {
+    console.warn(
+      `WARNING: While tracking your score PromptLayer had the following error: ${e}`
+    );
+    return false;
+  }
+  return true;
+};
+
+const promptLayerTrackPrompt = async (body: track.Prompt): Promise<boolean> => {
+  try {
+    const response = await fetch(
+      `${URL_API_PROMPTLAYER}/library-track-prompt`,
+      {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          ...body,
+          api_key: getApiKey(),
+        }),
+      }
+    );
+    const data = await response.json();
+    if (response.status !== 200) {
+      warnOnBadResponse(
+        data,
+        "WARNING: While tracking your prompt PromptLayer had the following error"
```

Comment:
"WARNING: While associating your request with a prompt template, PromptLayer experienced the following error"